src/app/multi-continuous-filter/multi-continuous-filter.component.ts

```diff
@@ -31,38 +33,44 @@ export class MultiContinuousFilterComponent extends StatefulComponent implements
     });
   }
 
-  public restoreContinuousFilter(state) {
+  public async restoreContinuousFilter(state) {
     if (!state['personFiltersState']) {
       return;
     }
-    let filters;
-    if(this.isFamilyFilter) {
-      filters = state['personFiltersState']['familyFilters'];
-    } else {
-      filters = state['personFiltersState']['personFilters'];
-    }
-    filters.forEach(filter => {
+    const filters = state['personFiltersState'][this.isFamilyFilter ? 'familyFilters' : 'personFilters'];
+    filters.forEach(async (filter) => {
```

Comment:
Remove for loop and leave an obnoxious comment detailing weird input data format